### ISSUE TITLE
Magit status commit (c c) error in terminal emacs

### ISSUE THREAD

**Issue by codedmart** [syl20bnr/spacemacs#4036]
This just started a day or two ago. When I am in magit status `SPC g s` and stage some files/hunks, if I try to commit `c c` it fails and I get an error in the *magit-process* buffer:

```
  1 git ... commit --
Waiting for Emacs
*ERROR*: Args out of range: [], 0
error: There was a problem with the editor '/usr/local/Cellar/emacs-mac/emacs-24.5-z-mac-5.13/bin/emacsclient --socket-name=/var/folders/8w/n_txhbn945j9gxp6bbz2nhkhr0000gn/T/emacs501/server'.
Please supply the message using either -m or -F option.
```

**Comment by StreakyCobra** [comment]
Have you tried to update the package to see if it goes away? Or was it because a packages update?

**Comment by codedmart** [comment]
Sorry forgot to mention that I am on `develop` branch and it start a day or two ago from a package update. I did revert back to `master` and can confirm that I can `c c` fine there. I just did another package update right before I reported this and the problem still persists in `develop`. Things do work as expected from GUI emacs.

**Comment by StreakyCobra** [comment]
Oh, actually I can confirm it on linux too.

```text
  1 git … commit --
Waiting for Emacs...
*ERROR*: Args out of range: [], 0
error: There was a problem with the editor '/usr/bin/emacsclient --socket-name=/tmp/emacs1000/server24611'.
Please supply the message using either -m or -F option.
```

***** 

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.0
- Spacemacs branch: contrib/bepo (rev. 40a63b9)
- Distribution: spacemacs
- Layers:
```elisp
(streakycobra asciidoc auto-completion better-defaults c-c++ colors dockerfile deft emacs-lisp emoji erc ess evil-cleverparens extra-langs finance git github haskell html ipython-notebook javascript latex markdown ocaml org pandoc python ranger restclient semantic shell shell-scripts
              (spell-checking :variables spell-checking-enable-auto-dictionary t)
              syntax-checking themes-megapack theming typography version-control yaml bepo)
```

**Comment by nixmaniack** [comment]
High Priority! Any known workarounds except from dropping to shell?

**Comment by StreakyCobra** [comment]
Does it happen with raw emacs with just magit?

**Comment by justbur** [comment]
Those don't look like backtraces. Has anyone tried to get one yet?

**Comment by justbur** [comment]
there was a recent change to with-editor here https://github.com/magit/magit/commit/310b5fa6387c18056a365409b9008092d4296006

**Comment by StreakyCobra** [comment]
@justbur There is no backtrace

**Comment by justbur** [comment]
Is this only with the daemon?

**Comment by StreakyCobra** [comment]
`*Messages*`:
```text
(No files need saving)
SPC g s
Type C-h i to view popup manual, ? to describe an argument or action.
Running git commit --
(No files need saving)
Running git commit --
There was a problem with the editor '/usr/bin/emacsclient --socket-name=/tmp/emacs1000/server14029'. ... [See buffer *magit-process: .spacemacs.d for details]
(No files need saving)
There was a problem with the editor '/usr/bin/emacsclient --socket-name=/tmp/emacs1000/server14029'. ... [See buffer *magit-process: .spacemacs.d for details]
Reverting (up to) 1 file-visiting buffer(s)...
(No buffers need to be reverted)
```

`*magit-process: …*`:
```text
  1 git … commit --
Waiting for Emacs...
*ERROR*: Args out of range: [], 0
error: There was a problem with the editor '/usr/bin/emacsclient --socket-name=/tmp/emacs1000/server14029'.
Please supply the message using either -m or -F option.
```

`SPC t D` changes nothing. I tried with `emacs -nw`, so not in daemon mode, I even stopped it to be sure.

**Comment by justbur** [comment]
looks like a magit bug to me

**Comment by StreakyCobra** [comment]
For me too, it's why I asked if someone can try to reproduce it without Spacemacs to be sure before reporting upstream :-)

**Comment by justbur** [comment]
@StreakyCobra you still get a reference to `emacsclient` when you use `emacs -nw`? That doesn't make sense

**Comment by StreakyCobra** [comment]
Yes, even with a fresh `.spacemacs`. Maybe something related to my shell configuration.

**Comment by codedmart** [comment]
I tried replacing `with-editor.el` with a earlier version but still see the same problem. I also get a reference to `emacsclient` when using `emacs -nw`.

**Comment by codedmart** [comment]
@justbur Can you not reproduce this error? Is all working fine for you in terminal emacs?

**Comment by justbur** [comment]
I use the gui version and haven't seen any changes

**Comment by codedmart** [comment]
It is only related to the terminal. I don't have the problem in the gui version.

**Comment by justbur** [comment]
> It is only related to the terminal. I don't have the problem in the gui version.

That suggests to me that it's something with the environment

**Comment by StreakyCobra** [comment]
@codedmart has it on OS X, @justbur has it on Windows, I have it on Linux, all with different setups, so it looks like an upstream bug.

**Comment by codedmart** [comment]
@justbur @StreakyCobra a `git bisect` resulted in 7e0d8f2d32fb8275a8fc044bb1396a7ef1664b78.

**Comment by codedmart** [comment]
That seems unrelated to me, but anyone have some ideas from here?

**Comment by codedmart** [comment]
Nevermind because of a different error my bisect failed. Let me try again.

**Comment by codedmart** [comment]
OK did a new `git bisect` and a19c0f3d2457912e80a7934e8060096a90a733c8 appears to be the culprit. I manually confirmed this as well. The commit just before a19c0f3d2457912e80a7934e8060096a90a733c8 is working as expected.

**Comment by StreakyCobra** [comment]
This one will be reverted by https://github.com/syl20bnr/spacemacs/pull/4025

**Comment by StreakyCobra** [comment]
Can you try again with the patch applied?

**Comment by codedmart** [comment]
@StreakyCobra @justbur Yes #4025 fixes the issue. Shall I close this now or does anyone need it open?

**Comment by StreakyCobra** [comment]
:+1: Let it open until the PR is merged :-) We will take care of closing it

**Comment by justbur** [comment]
That's so strange

**Comment by StreakyCobra** [comment]
@codedmart It should be fine now, can you test it?

**Comment by codedmart** [comment]
@StreakyCobra sorry got busy at the end of last week. I updated this morning and everything seems to be working as expected. Thanks!

**Comment by StreakyCobra** [comment]
:+1:
